### PR TITLE
[PM-14468] - add feature flag for critical apps

### PR DESCRIPTION
**Pull request by jaasen-livefront**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-14468

## 📔 Objective

This PR places all "Critical App" features behind a `CriticalApps` feature flag. This includes the following features:

- "Critical Apps" tab
- as well as the "Mark as Critical" button in the "All Applications" tab
- checkboxes in the "All Applications" tab table

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: apps/web/src/app/tools/risk-insights/all-applications.component.ts [hunk 1]
```diff
@@ -60,6 +63,10 @@ export class AllApplicationsComponent implements OnInit {
         }),
       )
       .subscribe();
+
+    this.isCritialAppsFeatureEnabled = await this.configService.getFeatureFlag(
+      FeatureFlag.CipherKeyEncryption,
```
Comment (djsmith85) [inline apps/web/src/app/tools/risk-insights/all-applications.component.ts:68]: ⚠️ This is requesting the wrong feature flag. Should be `CriticalApps`

File: apps/web/src/app/tools/risk-insights/risk-insights.component.ts [hunk 2]
```diff
@@ -65,9 +68,16 @@ export class RiskInsightsComponent {
     });
   };
 
+  async ngOnInit() {
+    this.isCritialAppsFeatureEnabled = await this.configService.getFeatureFlag(
+      FeatureFlag.CipherKeyEncryption,
```
Comment (djsmith85) [inline apps/web/src/app/tools/risk-insights/risk-insights.component.ts:73]: ⚠️ This is requesting the wrong feature flag. Should be `CriticalApps`

**Review by djsmith85 — CHANGES_REQUESTED** [review]

File: apps/web/src/app/tools/risk-insights/all-applications.component.ts (hunk 1 above)
Comment (jaasen-livefront) [reply apps/web/src/app/tools/risk-insights/all-applications.component.ts]: Nice catch! Fixed.

File: libs/common/src/enums/feature-flag.enum.ts [hunk 3]
```diff
@@ -35,6 +35,7 @@ export enum FeatureFlag {
   AccessIntelligence = "pm-13227-access-intelligence",
   Pm13322AddPolicyDefinitions = "pm-13322-add-policy-definitions",
   LimitCollectionCreationDeletionSplit = "pm-10863-limit-collection-creation-deletion-split",
+  CriticalApps = "critical-apps",
```
Comment (djsmith85) [inline libs/common/src/enums/feature-flag.enum.ts:38]: According to the [server PR](https://github.com/bitwarden/server/pull/4975/files#diff-94998af117f486ce85a16234b1bf36dd8c918651ff4d5c851927d203f5120e4eR151) this should be
```suggestion
  CriticalApps = "pm-14466-risk-insights-critical-application",
```

**Review by djsmith85 — CHANGES_REQUESTED** [review]

**Review by djsmith85 — APPROVED** [review]

**Pull request merged**